Sources/CircularProgress/CircularProgress.swift

```diff
@@ -415,3 +371,60 @@ public final class CircularProgress: NSView {
 		progressLabel.isHidden = !cancelButton.isHidden
 	}
 }
+
+extension CircularProgress:MouseTrackable{
+	public override func hitTest(_ point: NSPoint) -> NSView? {
+		guard let mousePos = self.window?.mouseLocationOutsideOfEventStream else {return nil}
+		let localMousePos = self.convert(mousePos,from:nil)
+		if let path = path {
+			let contains:Bool = path.contains(localMousePos)
+			return contains ? self : nil
+		}
```

Comment:
The logic for hit testing should reside in the `MouseTrackable` extension. It should handle all the hit logic and here we should only need to supply a `path` to be hit tested against.

Reply:
- The whole "why" this is here was removed when I simplified the comments. 
- Basically the way this is setup is a bit intricate. And it can't really be simplified anymore, I've tried a lot in the past. Like, you have to override hitTest in the class, because hit testing applies to the hierarchy. So if you have a covering NSView it needs to sort of run down the hierarchy. It's complicated. And I'm not explaining it well. 🤷
My comments when I use this solution more generically on other projects. Like for roundButtons
```swift
/**
    * This is the last NSView so we dont forward the hitTest to further descendants, however we could forward the hit test one more step to the CALayer
    * - TODO: ⚠️️ the logic inside this method should be in the Shape, and this method should just forward to the shape, if there is a shape
    */
```

Reply:
I feel this is best: Put the logic inside this method should be in the Shape, and this method should just forward to the shape, if there is a shape. 

But I feel the second best place is to put that code in the NSView where it is now. 

I feel, the code is messy, but other stuff can be cleaned up before this to make things clean. I can do that too, in another PR. If you want. 

Its up to you. If you want it in the protocol I will oblige. ✌️